.changeset/short-cycles-film.md

```diff
@@ -0,0 +1,8 @@
+---
+'@sphinx-labs/contracts': minor
+'@sphinx-labs/plugins': minor
```

Comment:
Before this PR, users commonly ran into errors like this at the end of the proposal process:
> Detected another project with the same owners, salt, and threshold but a different name. We currently do not support changing the name of an existing project. Please update your script to use the correct name:...

I noticed that we throw these types of error messages during project registration now, which is great. Now that this PR is implemented, are there any other errors that users could commonly hit at the end of the proposal process? I'm asking because this was one of the most frustrating aspects of the proposal process for me

Reply:
There are a number of checks that a user could theoretically encounter when the proposal request is sent to the backend. If you are interested, you can [review them all here](https://github.com/sphinx-labs/sphinx-managed/blob/pate/chu-634-require-explicit-project-registration-instead-of-implicitly/website/server/api/propose/index.ts#L561). 

In practice, most of these are checking for things that indicate a bug in our tooling rather than a user error. For example, we still check that the user proposed with a `projectName` and `safeAddress` that belong to the same project. This check should no longer be hit due to user error because the user only defines the project name in their script. It's on us to make sure we're using the correct Safe for that project. 

The only check the user realistically may hit due to their error is the [currently executing project check](https://github.com/sphinx-labs/sphinx-managed/blob/pate/chu-634-require-explicit-project-registration-instead-of-implicitly/website/server/api/propose/index.ts#L585). This error is thrown if they propose a deployment while another one is in the process of being executed.